### DISCUSSION TITLE
SAML authentication configuration to support multiple server configuration

### DISCUSSION THREAD

**Discussion by oguznsari** [keycloak/keycloak#28740] [Q&A]
Hi guys I am working on a project and we've Keycloak as an authentication provider.

We have a multiple-server configuration for redundancy while one is active the other server is in standby mode. 

When we use local login I can redirect either of my labs and it works (sessions created, authentication works smoothly).

Recently we've started to support external identity providers like SAML in this case, it works with only 1 server, and the authentication fails if I redirect to other server, please see the configuration below:

![image](https://github.com/keycloak/keycloak/assets/33628345/92b8add8-9e53-4ac5-8104-79cf02055235)

current SAML config:
![saml-c](https://github.com/keycloak/keycloak/assets/33628345/03745556-4cf6-46e4-b1ea-c904b4db76d2)

Tested scenario:
Initially only .128 were set and it is not working if I redirect it to .127
![test-condition](https://github.com/keycloak/keycloak/assets/33628345/6fb039f8-48d3-4ca2-afdd-801d3380fb9d)

As it can be shown my initial reaction was to add all possible servers to the Valid Redirect URIs but it failed. I don't know how to enable this or if it is even possible with Keycloak saml integration?

my result was failure with
"Could not send authentication request to identity provider."
the keycloak logs
[2209177_keycloak.log](https://github.com/keycloak/keycloak/files/14979653/2209177_keycloak.log)

**Comment by codespearhead** [comment]
I'm not sure I understand your setup: the authentication only works for the first entry of `Valid Redirect URIs`, but not the others, correct?

Also, by "multiple-server configuration for redundancy", do you mean you want to create a highly-available [Keycloak cluster](https://www.keycloak.org/high-availability/deploy-keycloak-kubernetes)?

**Reply by oguznsari** [reply]
thank you @codespearhead,

Yes, the authentication only works for the first entry of Valid Redirect URIs, but not the others
no, I don't want to create a high-available Keycloak cluster. I want only 1 Keycloak client to provide authentication for 2 different servers imagine Server A and Server B these are redundant to one another that's my setup and despite the local user login working SAML authentication doesn't work.
